GSoC-2018:  Initial project ideas, call for mentors and doc updates

oleg-nenashev opened the pull request:
I am going to make some edits tomorrow, but generally it is ready for review (especially the blogpost). 

- [x] GSoC subproject update (mentor page, landing)
- [x] GSoC 2018 project ideas
- [x] Call-for-mentors blog-post

<img width="1116" alt="screen shot 2018-01-05 at 01 24 18" src="https://user-images.githubusercontent.com/3000480/34590460-38812f84-f1b7-11e7-8681-1fec7f91936b.png">


@jenkins-infra/copy-editors @rtyler @batmat @christ66

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+:author: oleg-nenashev
+---
+
+This year we are interested to participate in
```
"interested in participating in"

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+== What is GSoC?
+
+link:https://developers.google.com/open-source/gsoc/[Google Summer of Code]
+(GSoC) is an annual, international, program which encourages
```
"annual international program"

oleg-nenashev replied:
I copy-pasted it from somewhere, will fix that

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+
+== What do you get?
+
+* A student who works within the area of your interest on full-time for several months
```
Strike out "on"

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+
+* A student who works within the area of your interest on full-time for several months
+* Joint projects with Jenkins experts, lots of fun and ability to study something together
+* Limited edition of swags from Google and Jenkins project
```
"Limited-edition swag"

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+* actively participate in the project during student selection, community bonding and coding phases (March - August)
+* work in teams of 2+ mentors per 1 each student
+* dedicate a consistent and significant amount of time, especially during the coding phase (_~5 hours_ per week in a team of two mentors)
+
```
Let's use caps at the start of each of these elements, it's consistent with the previous list and IMO the preferred style.

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+Outcome: decision whether we apply to GSoC in 2018.
+* Jan 21 - Application to GSoC (deadline - Jan 23)
+* Feb 12 - List of accepted mentoring organizations published
+* Next - If accepted, follow the link:https://developers.google.com/open-source/gsoc/timeline[GSoC Timeline] (if accepted)
```
"if accepted" appears twice, one can be removed.

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+
+== How to apply?
+
+If you are interested to propose a project or to join an existing one, respond to
```
"If you are interested in proposing a project or joining an existing one..."

omehegan-cb commented on content/projects/gsoc.adoc:
```diff
@@ -26,16 +20,33 @@ In exchange, numerous Jenkins community members volunteer as "mentors"
+Outcome: decision whether we apply to GSoC in 2018.
+* Jan 21 - Application to GSoC (deadline - Jan 23)
+* Feb 12 - List of accepted mentoring organizations published
+* Next - If accepted, follow the link:https://developers.google.com/open-source/gsoc/timeline[GSoC Timeline] (if accepted)
```
"If accepted" appears twice.

omehegan-cb commented on content/projects/gsoc/gsoc2018-project-ideas.adoc:
```diff
@@ -0,0 +1,157 @@
+as an opinionated way to configure Jenkins based on human-readable declarative configuration files (YAML).
+Plugin should be easy to use regardless of previous Jenkins experience or Jenkins model expert knowledge.
+
+This will allow to fully automate Jenkins deployment process and manage Jenkins (one or multiple instances) configuration.
```
"This will allow Jenkins admins to..."

omehegan-cb commented on content/projects/gsoc/gsoc2018-project-ideas.adoc:
```diff
@@ -0,0 +1,157 @@
+* Cloud-based storage support (link:https://groups.google.com/d/msg/jenkinsci-dev/z40kn8IqFb8/YkdgbuScCgAJ[discussion])
+* Integration with the core's “Discard old builds” feature
+* Improvements to data retention policies (e.g. link:https://issues.jenkins-ci.org/browse/JENKINS-48715[JENKINS-48715])
+workspace usage statistics
```
Looks like this line is superfluous?

omehegan-cb commented on content/projects/gsoc/gsoc2018-project-ideas.adoc:
```diff
@@ -0,0 +1,157 @@
+
+The link:https://github.com/jenkinsci/acceptance-test-harness[Jenkins Acceptance Test Harness (ATH)] is a great vehicle
+to test Jenkinsfiles and custom DSL libraries ahead of deploying them to production servers.
+However, it has couple of drawbacks.
```
Combine this at the end of the last sentence, e.g., "...deploying them to production servers, but it has a couple of drawbacks."

omehegan-cb commented on content/projects/gsoc/gsoc2018-project-ideas.adoc:
```diff
@@ -0,0 +1,157 @@
+to test Jenkinsfiles and custom DSL libraries ahead of deploying them to production servers.
+However, it has couple of drawbacks.
+
+* it can be quite slow as it needs to bootstrap an entire Jenkins instance for each test method.
```
Caps at the start of each list item.

omehegan-cb commented on content/projects/gsoc/gsoc2018-project-ideas.adoc:
```diff
@@ -0,0 +1,157 @@
+However, it has couple of drawbacks.
+
+* it can be quite slow as it needs to bootstrap an entire Jenkins instance for each test method.
+* real production environments typically need to use a very specific plugin list of pre-defined plugins and plugin versions
```
"very specific list of pre-defined plugins and their versions" would be a little more concise.

omehegan-cb commented on content/projects/gsoc/mentors.adoc:
```diff
@@ -0,0 +1,66 @@
+** The mentorship work should be performed during spare time or during the OSS contribution time dedicated by the company.
+In the latter case the mentor should ensure that there will be a consistent time dedicated over the entire GSoC mentorship period
+** The project proposed by mentors should not overlap neither with direct responsibilities within the company nor with the company product roadmaps.
+** He(she) should ensure there is no conflict of interest between the project and the work responsibilities
```
He/She

omehegan-cb commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,92 @@
+== What is GSoC?
+
+link:https://developers.google.com/open-source/gsoc/[Google Summer of Code]
+(GSoC) is an annual, international, program which encourages
```
The 'Google Summer of Code' will be the link text here, as shown in the brackets (this threw me off at first too!).

oleg-nenashev commented:
@omehegan-cb @MarkEWaite Addressed all feedback outside the project ideas section. Regarding them, I will contact the submitters. For now I have added a not which says that the ideas a re in the draft state

bitwiseman commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,93 @@
+If you are interested in proposing a project or joining an existing one, please respond to
+link:https://groups.google.com/forum/#!topic/jenkinsci-dev/We-14-z_YXU[this thread]
+in the Jenkins Developer mailing list.
+We kindly ask to do it before *Jan 17, 2017* so that we could add it to the application materials.
```
ask that you to do it before *Jan 17, 2017* so that we can add

bitwiseman commented on content/blog/2018/01/2018-01-05-gsoc2018-call-for-mentors.adoc:
```diff
@@ -0,0 +1,93 @@
+link:https://docs.google.com/document/d/1q2p_XZEdbkcVDMpEPTtjPS15i2Oq3CQgH_geJjPhofY/edit[this document].
+In this document you need to briefly describe your project idea and required qualifications from students
+
+Project requirements
```
Should this be a heading?

oleg-nenashev commented:
I will adjust the date and merge it

On Jan 6, 2018 00:36, "Liam Newman" <notifications@github.com> wrote:

> *@bitwiseman* approved this pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/1310#pullrequestreview-87051553>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC3IoKGzROw_ja3KFcaYu_vCNkcI0o8Pks5tHrHmgaJpZM4RT4jf>
> .
>

oleg-nenashev commented:
🚢 🇮🇹 , so that @orrc can post his post on Monday
